### PR TITLE
Add 0.205 release notes

### LINKED ISSUE #10828: Release notes for 0.205
Issue to track release notes for next release

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Fixes #10828.

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+* Add support for prepared statements.
+* Add partial query cancellation via ``partialCancel()`` on ``PrestoStatement``.
+* Use ``VARCHAR`` rather than ``LONGNVARCHAR`` for the Presto ``varchar`` type.
+* Use ``VARBINARY`` rather than ``LONGVARBINARY`` for the Presto ``varbinary`` type.
+
+Hive Connector Changes
+----------------------
+
+* Improve the query performance when writing static partitions.
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: @wenleix Please clarify the first two items in the Hive changes here.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  of ``concurrent_lifespans_per_task`` session property.
+* Add ``query_max_total_memory`` session property and ``query.max-total-memory`` config
+  property. Queries will be aborted after their total (user + system) memory reservation
+  exceeds this threshold.
+* Improve the error message when access is denied when selecting from a view due to the
+  view owner having insufficient permissions to create the view.
+* Improve stats calculation for outer joins and correlated subqueries.
+* Change ``DELETE FROM <table> WHERE <condition>`` to check that the user has ``SELECT``
+  privileges on the ``WHERE`` condition, which is more in line with the ANSI standard.
+* Reduce memory usage for columns of all ``NULL`` or all non-null values.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: This isn't only for columns which contain all-nulls or all-non-nulls. As this is on the block level, this actually improves whenever there are long sequences or all-nulls or all-non-nulls.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+* Change the internal hash function used in  ``approx_distinct``. For a fixed value set,
+  result of ``approx_distinct`` may change in this version compared to the previous
+  version. The standard error of the results should still be within the configured bounds.
+* Improve efficiency and reduce memory usage for scalar correlated subqueries with aggregation.
+* Remove the legacy local scheduler and associated configuration properties.
+* Disallow ``FILTER`` clause for ``IF``, ``COALESCE``, and ``NULLIF`` functions.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: i would call it "Reject", since it was never "allowed".

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+* Disallow having multiple public constructors in ``@ScalarFunction`` classes. All non-static
+  implementations of a SQL function will now be associated with a single constructor.
+  This fixes expressivity gap of providing specialized implementations for SQL functions
+  with generic arguments.
+
+JDBC Driver Changes
+-------------------
+
+* Add support for prepared statements.
+* Add partial query cancellation via ``partialCancel()`` on ``PrestoStatement``.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: To me it wasn't obvious what that means.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+General Changes
+---------------
+
+* Fix parsing of row type signatures with spaces and types containing spaces.
+* Fix distributed planning failure for complex queries when using bucketed execution.
+* Fix a bug that triggers checks for ``SELECT`` privileges for tables in ``DELETE`` queries.
+  This bug only affects connectors that implement ``checkCanSelectFromColumns()``.
+* Fix a bug in enforcing column access controls when querying views. This bug
+  only affects connectors that implement the column access control methods.
+* Fix a bug in :func:`ST_ExteriorRing` to make it only work with polygons.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: Avoid "fix a bug". How about (not sure I have the terminology correct)
```
* Fix :func:`ST_ExteriorRing` to only accept polygons.
  Previously, it erroneously accepted other geometries.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+* Add the ``task.min-drivers-per-task`` and ``task.max-drivers-per-task`` config options.
+  The former specifies the guaranteed minimum number of drivers a task will run concurrently
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: Change the sentences to be consistent:

*"a task will run"* vs *"that can run ... for a task"*

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  given that it has enough work to do. The latter specifies the maximum number of drivers
+  that can run concurrently for a task.
+* Add ``concurrent-lifespans-per-task`` config property to control the default value
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]:
```
* Add the ``concurrent-lifespans-per-task`` config property to control the default value
  of the ``concurrent_lifespans_per_task`` session property.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  of ``concurrent_lifespans_per_task`` session property.
+* Add ``query_max_total_memory`` session property and ``query.max-total-memory`` config
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]:
```
* Add the ``query_max_total_memory`` session property and the ``query.max-total-memory``
  config property. Queries will be aborted after their total (user + system) memory
  reservation exceeds this threshold.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  property. Queries will be aborted after their total (user + system) memory reservation
+  exceeds this threshold.
+* Improve the error message when access is denied when selecting from a view due to the
+  view owner having insufficient permissions to create the view.
+* Improve stats calculation for outer joins and correlated subqueries.
+* Change ``DELETE FROM <table> WHERE <condition>`` to check that the user has ``SELECT``
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: Move to security changes section

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  privileges on the ``WHERE`` condition, which is more in line with the ANSI standard.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: "more in line" is weak and seems to imply that we are not strictly following the standard. Should we change the second part to *"as is required by the SQL standard"*?

(it's now "SQL standard" rather than "ANSI", now that it's an ISO standard)

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+* Reduce memory usage for columns of all ``NULL`` or all non-null values.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: Write "null" in lowercase, to be consistent with the non-null part

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+* Change the internal hash function used in  ``approx_distinct``. For a fixed value set,
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]:
```
* Change the internal hash function used in  ``approx_distinct``. The result of ``approx_distinct``
  may change in this version compared to the previous version for the same set of values. However,
  the standard error of the results should still be within the configured bounds.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+  result of ``approx_distinct`` may change in this version compared to the previous
+  version. The standard error of the results should still be within the configured bounds.
+* Improve efficiency and reduce memory usage for scalar correlated subqueries with aggregation.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: "with aggregations"

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+* Remove the legacy local scheduler and associated configuration properties.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: We should mention the properties if we expect that people will need to delete them from their configurations

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+* Reject ``FILTER`` clause for ``IF``, ``COALESCE``, and ``NULLIF`` functions.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]:
```
* Do not allow using the ``FILTER`` clause for the ``COALESCE``, ``IF``, or ``NULLIF`` functions.
  It was previously allowed but was ignored.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+* Disallow non-static ``@ScalarFunction``-annotated methods. Non-static SQL function
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: Move to SPI Changes section

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: Let's write this as
```
* Disallow non-static methods to be annotated with ``@ScalarFunction``. Non-static SQL function
  implementations must now be declared in a class annotated with ``@ScalarFunction``.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+  implementations must now be declared in ``@ScalarFunction``-annotated classes.
+* Disallow having multiple public constructors in ``@ScalarFunction`` classes. All non-static
+  implementations of a SQL function will now be associated with a single constructor.
+  This fixes expressivity gap of providing specialized implementations for SQL functions
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: "This improves support for providing specialized implementations of SQL functions with generic arguments."

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 4 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: I agree, though this is a feature of the Presto protocol that should be better documented elsewhere (and unfortunately isn't)

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 1 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: Remove "the"

What does "static partition" mean? We don't have that concept in Presto (all writes are "dynamic" in Hive terms)

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+* Add partial query cancellation via ``partialCancel()`` on ``PrestoStatement``.
+* Use ``VARCHAR`` rather than ``LONGNVARCHAR`` for the Presto ``varchar`` type.
+* Use ``VARBINARY`` rather than ``LONGVARBINARY`` for the Presto ``varbinary`` type.
+
+Hive Connector Changes
+----------------------
+
+* Improve the query performance when writing static partitions.
+* Improve stripe size estimation for the optimized ORC writer.
+  This reduces cases when writing tiny ORC stripes.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: I think this is trying to say
```
This reduces the number of cases where tiny ORC stripes must be written.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+* Add support for the ``skip.footer.line.count`` table property.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: Let's change this to
```
* Respect the ``skip.footer.line.count`` Hive table property.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+
+CLI Changes
+-----------
+
+* Fix CLI to properly support some 256 color terminals.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: Is this "properly support"? I thought it crashed previously?
```
* Properly support some 256 color terminals.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+
+SPI Changes
+-----------
+
+* Add a context parameter to the ``create()`` method in ``SessionPropertyConfigurationManagerFactory``.
+  This is a backwards incompatible change with the previous connector SPI.
+  If you have written a session property configuration manager, you will
+  need to update your code before deploying this release.
+* Deprecate ``checkCanSelectFromTable/View`` and ``checkCanCreateViewWithSelectFromTable/View`` in the AccessControl SPI.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]:
```
* Deprecate ``checkCanSelectFrom{Table,View}`` and ``checkCanCreateViewWithSelectFrom{Table,View}``
  in ``ConnectorAccessControl`` and ``SystemAccessControl``. ``checkCanSelectFromColumns``
  and ``checkCanCreateViewWithSelectFromColumns`` should be used instead.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 22]
```diff
@@ -0,0 +1,72 @@
+=============
+Release 0.205
+=============
+
+General Changes
+---------------
+
+* Fix parsing of row type signatures with spaces and types containing spaces.
+* Fix distributed planning failure for complex queries when using bucketed execution.
+* Fix a bug that triggers checks for ``SELECT`` privileges for tables in ``DELETE`` queries.
```
[the remark this answers is not in the document]
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: /cc @rschlussel

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+  This bug only affects connectors that implement ``checkCanSelectFromColumns()``.
+* Fix a bug in enforcing column access controls when querying views. This bug
```
[the remark this answers is not in the document]
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: /cc @rschlussel

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 5 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: /cc @jwass @mbasmanova

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 14 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: /cc @raghavsethi

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 1 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: /cc @wenleix

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 18 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: /cc @wenleix please provide more details.

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 20 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: I don't have all the context, it was added to the release notes issues as`* Fix Presto CLI to do not throw the NumberFormatException for some 256 color terminals`, which is not user friendly. If it was crashing we should reword this. /cc @arhimondr

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 2 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: @electrum Since @dain is out, are you familiar with this change? Should we reword this as @findepi proposes?

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 24]
```diff
@@ -0,0 +1,72 @@
+=============
+Release 0.205
+=============
+
+General Changes
+---------------
+
+* Fix parsing of row type signatures with spaces and types containing spaces.
```
[the remark this answers is not in the document]
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: /cc @caithagoras

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 5 above)
Comment (mbasmanova) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: @electrum @nezihyigitbasi I like David's version.

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 22 above)
[the remark this answers is not in the document]
Comment (rschlussel2) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: We no longer check for ``SELECT` privileges for tables in ``DELETE`` queries.  ``SELECT`` privileges are not required, so the user impact was that people's queries could have failed unnecessarily if they had ``DELETE`` privileges, but not ``SELECT`` privileges.

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 1 above)
Comment (wenleix) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: @electrum : Yes. But we will use constant values for partition values. And we optimize for such case (so don't need to lookup in HashTable per column)

Comment (wenleix) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: @electrum : We can say 
```
Improve the query performance when all partition column values are constants.
```

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 23 above)
[the remark this answers is not in the document]
Comment (rschlussel2) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: When selecting from a view, check that the view owner had permission to create the view.

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 22 above)
[the remark this answers is not in the document]
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: How about:

```
* Remove unnecessary check for ``SELECT`` privileges for ``DELETE`` queries.
  Previously, ``DELETE`` queries could fail if the user only has ``DELETE``
  privileges, and not the ``SELECT`` privileges.
```

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 18 above)
Comment (wenleix) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]:
```
This reduces the number of cases where tiny ORC stripes will be written.
```

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 23 above)
[the remark this answers is not in the document]
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: How about:

```
* Add a check to verify that the view owner had permission to create the view when
  running ``SELECT`` queries against a view.
```

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 24 above)
[the remark this answers is not in the document]
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: This allow users to create columns/expressions with types such as `row(timestamp with time zone)`. For example, allow user to do casts:
```
cast(row(timestamp '2018-06-01') as row(timestamp with time zone))
```.
Before the fix, this cast will fail since `timestamp` is treated as field name and type `with time zone` does not exists

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 14 above)
Comment (raghavsethi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: task.legacy-scheduling-behavior and task.level-absolute-priority

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 2 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: @electrum @nezihyigitbasi this relates to https://github.com/prestodb/presto/pull/10709 and https://github.com/prestodb/presto/pull/10842

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 25]
```diff
@@ -0,0 +1,86 @@
...
+  The syntax was previously allowed but was otherwise ignored.
+
+Security Changes
+----------------
+
+* Remove unnecessary check for ``SELECT`` privileges for ``DELETE`` queries.
+  Previously, ``DELETE`` queries could fail if the user only has ``DELETE``
+  privileges but not ``SELECT`` privileges.
+  This only affected connectors that implement ``checkCanSelectFromColumns()``.
+* Add a check that the view owner has permission to create the view when
```
Comment (rschlussel2) [inline presto-docs/src/main/sphinx/release/release-0.205.rst]: This only affects connectors that implement `checkCanCreateViewWithSelectFromColumns()`.   (we were doing the right thing for `checkCanCreateViewWithSelectFromTable()`)

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+  running ``SELECT`` queries against a view. This only affected connectors that
+  implement ``checkCanCreateViewWithSelectFromColumns()``.
+* Change ``DELETE FROM <table> WHERE <condition>`` to check that the user has ``SELECT``
+  privileges on the ``WHERE`` condition as is required by the SQL standard.
```
[the remark this answers is not in the document]
Comment (rschlussel2) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: exactly.  For a query that says `DELETE FROM table` we only check delete privileges. If the query is `DELETE FROM table WHERE x > 0` we also check select privileges on x.

File: presto-docs/src/main/sphinx/release/release-0.205.rst [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+
+* Add support for prepared statements.
+* Add partial query cancellation via ``partialCancel()`` on ``PrestoStatement``.
+* Use ``VARCHAR`` rather than ``LONGNVARCHAR`` for the Presto ``varchar`` type.
+* Use ``VARBINARY`` rather than ``LONGVARBINARY`` for the Presto ``varbinary`` type.
+
+Hive Connector Changes
+----------------------
+
+* Improve query performance when all partition column values are constants.
```
[the remark this answers is not in the document]
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: Yes, will update shortly.

File: presto-docs/src/main/sphinx/release/release-0.205.rst (hunk 26 above)
[the remark this answers is not in the document]
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-0.205.rst]: How about we change this to
```
* Change ``DELETE FROM <table> WHERE <condition>`` to check that the user has ``SELECT``
  privileges on the objects referenced by the ``WHERE`` condition as is required by the SQL standard.
